Source/WebKit/WebProcess/Plugins/PDF/PDFPluginBase.mm

```diff
@@ -197,6 +200,247 @@
     return m_rootViewToPluginTransform.mapPoint(point);
 }
 
+void PDFPluginBase::updateControlTints(GraphicsContext& graphicsContext)
+{
+    ASSERT(graphicsContext.invalidatingControlTints());
+
+    if (m_horizontalScrollbar)
+        m_horizontalScrollbar->invalidate();
+    if (m_verticalScrollbar)
+        m_verticalScrollbar->invalidate();
+    invalidateScrollCorner(scrollCornerRect());
+}
+
+IntRect PDFPluginBase::scrollCornerRect() const
+{
+    if (!m_horizontalScrollbar || !m_verticalScrollbar)
+        return IntRect();
+    if (m_horizontalScrollbar->isOverlayScrollbar()) {
+        ASSERT(m_verticalScrollbar->isOverlayScrollbar());
+        return IntRect();
+    }
+    return IntRect(m_view->width() - m_verticalScrollbar->width(), m_view->height() - m_horizontalScrollbar->height(), m_verticalScrollbar->width(), m_horizontalScrollbar->height());
+}
+
+ScrollableArea* PDFPluginBase::enclosingScrollableArea() const
+{
+    // FIXME: Walk up the frame tree and look for a scrollable parent frame or RenderLayer.
```

Comment:
Also PDFPluginBase has no business knowing about its ancestor tree, so it should call out to some other class to do this ancestor walk.

Reply:
But at the same time, we've gotten away without implementing this in PDFPlugin for many years, so maybe we can continue? I don't know what it's for, guess I'll have to check.

Reply:
Oh! Maybe this is why smooth keyboard scrolling doesn't work.

Reply:
oh ya this would definitely break it too